dev/ruff/main.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/dagger/dagger/dev/ruff/internal/dagger"
+)
+
+var (
+	pythonVersion     = "3.11"
+	pythonImageRepo   = "docker.io/library/python"
+	pythonImageTag    = fmt.Sprintf("%s-slim", pythonVersion)
+	pythonImageDigest = "sha256:fc39d2e68b554c3f0a5cb8a776280c0b3d73b4c04b83dbade835e2a171ca27ef"
+	pythonImage       = pythonImageRepo + ":" + pythonImageTag + "@" + pythonImageDigest
```

Comment:
I don't think you need Python to use Ruff. A better installation method is to use their OCI image. See end of [Installing Ruff](https://docs.astral.sh/ruff/installation/):

> On **Docker**, it is published as `ghcr.io/astral-sh/ruff`, tagged for each release and `latest` for the latest release.
> ```console
> docker run -v .:/io --rm ghcr.io/astral-sh/ruff check
> docker run -v .:/io --rm ghcr.io/astral-sh/ruff:0.3.0 check
> ```

Reply:
This is directly derived from our current CI code. (`dev/sdk_python.go`). I am 100% on board with the improvement, but in the spirit of the "spin out then improve" approach, OK to make it a follow-up PR?

Reply:
Yes, of course. I already said as much in the summary 🙂